zx/coerce.go

```diff
@@ -148,9 +245,9 @@ func CoerceToPort(in zng.Value) (uint32, bool) {
 		out = uint32(v)
 	case zng.IdUint64:
 		var v uint64
-		v, err = zng.DecodeCount(body)
+		v, err = zng.DecodeUint(body)
 		// check for overflow
-		if v > math.MaxInt64 {
+		if v > math.MaxInt16 {
```

Comment:
Would you explain why MaxInt16 is the limit? Also, why this uint64 is cast down to a uint32 before returning from the function as a uint64?

Reply:
Whoops, should be MaxUint16 for a port, will fix.  I don't understand the second part of the question, the return value is a uint32 (which appears wrong to me, it should be a uint16 but I don't think that needs to be addressed in this PR)